mission/gui_auv/auv_gui/auv_gui.py

```diff
@@ -0,0 +1,283 @@
+import math
+import sys
+from threading import Thread
+from typing import List, Optional
+
+import matplotlib.pyplot as plt
+import numpy as np
+import rclpy
+from matplotlib.backends.backend_qt5agg import FigureCanvasQTAgg as FigureCanvas
+from nav_msgs.msg import Odometry
+from PyQt5.QtCore import QTimer
+from PyQt5.QtWidgets import (
+    QApplication,
+    QGridLayout,
+    QLabel,
+    QMainWindow,
+    QTabWidget,
+    QVBoxLayout,
+    QWidget,
+)
+from rclpy.executors import MultiThreadedExecutor
+from rclpy.node import Node
+from std_msgs.msg import Float32
+
+# --- Quaternion to Euler angles ---
+
+
+def quaternion_to_euler(x: float, y: float, z: float, w: float) -> List[float]:
+    """
+    Convert a quaternion to Euler angles (roll, pitch, yaw).
+
+    Args:
+        x (float): The x component of the quaternion.
+        y (float): The y component of the quaternion.
+        z (float): The z component of the quaternion.
+        w (float): The w component of the quaternion.
+
+    Returns:
+        List[float]: A list of Euler angles [roll, pitch, yaw].
+    """
+
+    # Roll (x-axis rotation)
+    sinr_cosp = 2 * (w * x + y * z)
+    cosr_cosp = 1 - 2 * (x * x + y * y)
+    roll = np.degrees(math.atan2(sinr_cosp, cosr_cosp))
+
+    # Pitch (y-axis rotation)
+    sinp = 2 * (w * y - z * x)
+    pitch = np.degrees(math.asin(sinp))
+
+    # Yaw (z-axis rotation)
+    siny_cosp = 2 * (w * z + x * y)
+    cosy_cosp = 1 - 2 * (y * y + z * z)
+    yaw = np.degrees(math.atan2(siny_cosp, cosy_cosp))
+
+    return [roll, pitch, yaw]
+
+
+# --- GUI Node ---
+
+
+class GuiNode(Node):
+    """ROS2 Node that subscribes to odometry data and stores x, y positions."""
+
+    def __init__(self) -> None:
+        """Initialize the GuiNode and set up the odometry subscriber."""
+        super().__init__("auv_gui_node")
+
+        # Subscriber to the /nucleus/odom topic
+        self.subscription = self.create_subscription(Odometry, '/nucleus/odom', self.odom_callback, 10)
```

Comment:
Add the topic names to a yaml config file, such that it is easy to change (we will have to change in the future)